engine/access/rpc/engine.go

```diff
@@ -128,11 +130,22 @@ func New(log zerolog.Logger,
 	// wrap the unsecured server with an HTTP proxy server to serve HTTP clients
 	httpServer := NewHTTPServer(unsecureGrpcServer, config.HTTPListenAddr)
 
+	cacheSize := config.ConnectionPoolSize
+	if cacheSize == 0 {
+		cacheSize = backend.DefaultConnectionPoolSize
+	}
+	cache, err := lru.New(cacheSize)
+	if err != nil {
+		log.Err(err).Msg("could not initialize connection pool cache")
+		return nil, err
```

Comment:
You can just wrap the err with context and return it, then let the caller log it.

if you unravel the calls, what will happen is the RPC engine returns the error to the component factory method, which then returns it to the builder, which finally throws a fatal error. So in the end, you'll get the error log message and fatal error for free